src/basics.ml

```diff
@@ -100,6 +100,10 @@ let eq : term -> term -> bool = fun a b -> a == b ||
     | (Prod(a1,b1), Prod(a2,b2))
     | (Abst(a1,b1), Abst(a2,b2)) -> let (_, b1, b2) = Bindlib.unbind2 b1 b2 in
                                     eq ((a1,a2)::(b1,b2)::l)
+    | (LLet(t1,a1,u1), LLet(t2,a2,u2)) ->
+        let (_, u1, u2) = Bindlib.unbind2 u1 u2 in
+        (* FIXME or equality of substituted? *)
```

Comment:
No, what you are doing is correct. This is not conversion, but syntactic equality (up to α-equivalence).